### PR TITLE
Add support for local user

### PR DESCRIPTION
**Pull request by vthiebaut10**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/ssh/azext_ssh/_help.py [hunk 1]
```diff
@@ -12,7 +12,8 @@
 
 helps['ssh vm'] = """
     type: command
-    short-summary: SSH into Azure VMs using an ssh certificate
+    short-summary: SSH into Azure VMs
```
Comment (bagajjal) [inline src/ssh/azext_ssh/_help.py:15]: I wouldn't touch the short-summary. We want to promote customers to use AAD based login. This short summary helps to emphasize that.

File: src/ssh/azext_ssh/custom.py [hunk 2]
```diff
@@ -79,11 +80,17 @@ def _do_ssh_op(cmd, resource_group, vm_name, ssh_ip, public_key_file, private_ke
 
         raise azclierror.ResourceNotFoundError(f"VM '{vm_name}' does not have a public or private IP address to SSH to")
 
-    public_key_file, private_key_file, delete_keys = _check_or_create_public_private_files(public_key_file,
+    # If user ptovides local user, no credentials should be deleted.
```
Comment (bagajjal) [inline src/ssh/azext_ssh/custom.py:83]: typo.
provides

File: src/ssh/azext_ssh/custom.py [hunk 3]
```diff
@@ -79,11 +80,17 @@ def _do_ssh_op(cmd, resource_group, vm_name, ssh_ip, public_key_file, private_ke
...
+    # If user provides a local user, use the provided credentials for authentication
+    if not username:
+        delete_cert = True
+        public_key_file, private_key_file, delete_keys = _check_or_create_public_private_files(public_key_file,
                                                                                            private_key_file,
                                                                                            credentials_folder)
-    cert_file, username = _get_and_write_certificate(cmd, public_key_file, None)
-    op_call(ssh_ip, username, cert_file, private_key_file, delete_keys)
+        cert_file, username = _get_and_write_certificate(cmd, public_key_file, None)
+    op_call(ssh_ip, username, cert_file, private_key_file, delete_keys, delete_cert)
```
Comment (bagajjal) [inline src/ssh/azext_ssh/custom.py:93]: Add a new line when the function  ends. It helps to improve code readability.

File: src/ssh/azext_ssh/ssh_utils.py [hunk 4]
```diff
@@ -19,47 +19,50 @@
...
 
 
-def start_ssh_connection(port, ssh_args, ip, username, cert_file, private_key_file, delete_keys):
+def start_ssh_connection(port, ssh_args, ip, username, cert_file, private_key_file, delete_keys, delete_cert):
 
     ssh_arg_list = []
     if ssh_args:
         ssh_arg_list = ssh_args
+    
+    print(cert_file)
```
Comment (bagajjal) [inline src/ssh/azext_ssh/ssh_utils.py:28]: delete

File: src/ssh/azext_ssh/ssh_utils.py [hunk 5]
```diff
@@ -19,47 +19,50 @@
...
-    _do_cleanup(delete_keys, cert_file, private_key_file)
-    if log_file:
-        file_utils.delete_file(log_file, f"Couldn't delete temporary log file {cert_file}. ", True)
-    if delete_keys:
-        temp_dir = os.path.dirname(cert_file)
-        file_utils.delete_folder(temp_dir, f"Couldn't delete temporary folder {temp_dir}", True)
+        # Make sure all files have been properly removed.
+        _do_cleanup(delete_keys, delete_cert, cert_file, private_key_file)
+        if log_file:
+            file_utils.delete_file(log_file, f"Couldn't delete temporary log file {cert_file}. ", True)
```
Comment (bagajjal) [inline src/ssh/azext_ssh/ssh_utils.py:62]: This should be outside of the if block..
We must delete log file even if both delete_cert, delete_keys are false.

**Review by bagajjal — APPROVED** [review]

File: src/ssh/azext_ssh/ssh_utils.py (hunk 5 above)
Comment (vthiebaut10) [reply src/ssh/azext_ssh/ssh_utils.py:62]: @bagajjal  We don't create a temporary log file when both delete_cert and delete_keys are false.

**Pull request merged**